### PR TITLE
mappedPlayerGuids to improve player search by GUID

### PR DESCRIPTION
**Pull request by danyelvarejao1**

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -473,12 +473,11 @@ Player* Game::getPlayerByGUID(const uint32_t& guid)
 		return nullptr;
 	}
 
-	for (const auto& it : players) {
-		if (guid == it.second->getGUID()) {
-			return it.second;
-		}
+	auto it = mappedPlayerGuids.find(guid);
```
Comment (Humberd) [inline src/game.cpp]: What if `Game::getPlayerByNameWildcard` had not been called before. It would not find a player, wouldn't it?

Comment (kygov) [reply src/game.cpp]: It's not related, in 'Game::addPlayer(Player* player)' is inserted player to 'mappedPlayerGuids' map, exactly same way as 'getPlayerByName' or 'getCreatureByName' etc.

Comment (danyelvarejao1) [reply src/game.cpp]: Yes, I have no problem with that, as mappedPlayersName is also added.

Comment (Lyuzera) [reply src/game.cpp]: this is an extra cache for improve seach by GUID as the title says. Now `Player(guid)` is fast as `Player(name)` or `Player(cid)`.

**Comment by Znote** [comment]
This is great, as player GUID is frequently used to index player data in locally, and sometimes complex structures of data in various script systems. The player userdata is also frequently reinitialized when passed through addEvent calls for delayed execution.

**Comment by nekiro** [comment]
Did you do any benchmarks if that changes anything?

**Comment by danyelvarejao1** [comment]
> Did you do any benchmarks if that changes anything?
Sure this is much faster.

**Comment by nekiro** [comment]
> @nekiro, Sure this is much faster.

I actually asked about some type of proof, data or something.

**Comment by kygov** [comment]
@nekiro 
I've made some tests, 'find' based function is almost **33x faster** than 'loop' on my pc.
Code I've used: https://pastebin.com/hUaMqRMq

> 1500 players size, 1500 mapped size
For Took 12446 ms
Find Took 366 ms

**Pull request merged**